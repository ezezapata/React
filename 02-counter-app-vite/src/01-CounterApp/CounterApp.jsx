import { useState } from 'react'
import PropTypes from 'prop-types';


export const CounterApp = ({value}) => {
    const [counter, setCounter] = useState( value );
    const handleAdd = () => { 
        setCounter(counter + 1);
    }
     
    return(
        <>
            <h1>CounterApp</h1>
            <p> {counter} </p>

            <button onClick={handleAdd}>
                +1
            </button>
        </>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
}

CounterApp.defaultProps = {
    value: 0,
}