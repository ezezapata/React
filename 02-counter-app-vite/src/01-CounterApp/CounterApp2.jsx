import PropTypes from 'prop-types';
import { useState } from 'react';

export const CounterApp2 = ({value}) => {
    const [counter, setCounter] = useState(value);

    const handleAdd = () => {
        setCounter(counter + 2);
    }
    
    return(<>
        <h1>CounterApp 2</h1>
        <p> {counter} </p>
        <button onClick={handleAdd}>
            +2
        </button>
    </>)
}

CounterApp2.propTypes = {
    value: PropTypes.number.isRequired,
}

CounterApp2.defaultProps = {
    value: 0,
}