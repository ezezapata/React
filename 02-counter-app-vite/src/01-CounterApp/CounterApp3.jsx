import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp3 = ({number}) => {
    // Por ahora, cuando cambia el estado, el componente se vuelve a ejecutar. Hay formas de evitar esto, lo voy a ver a futuro
    // console.log('render');

    const [counter, setCounter] = useState(number);

    const handleSubstract = () => {
        setCounter(counter - 1);
    }

    const handleAdd = () => {
        setCounter(counter + 1);
    }

    const handleReset = () => {
        setCounter(number);
    }

    return(
        <>
            <h1>CounterApp 3</h1>
            <p> {counter} </p>
            <button onClick={handleAdd}>+1</button>
            <button onClick={handleSubstract}>-1</button>
            <button aria-label='btn-reset' onClick={handleReset}>Reset</button>
        </>
    )
}

CounterApp3.propTypes = {
    number: PropTypes.number.isRequired,
}