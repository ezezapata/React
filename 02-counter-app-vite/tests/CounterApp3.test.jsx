import { fireEvent, render, screen } from '@testing-library/react';
import { CounterApp3 } from '../src/01-CounterApp/CounterApp3';


describe('Pruebas en CounterApp3', () =>{
    const number = 10;
    test('Debe hacer match con el snapshot.', () => {
        const { container } = render(<CounterApp3 number={ number }/>);
        expect( container ).toMatchSnapshot();
    });

    test('Debe mostrar el valor inicial de 100', () => {
        render( <CounterApp3 number={number}  /> );

        // El screen es la representación del componente renderizado
        expect( screen.getByRole('paragraph').innerHTML ).toContain( '10' );
    });

    test('Debe incrementar con el botón +1', () => {
        render( <CounterApp3 number={number} /> );
        fireEvent.click(screen.getByText( '+1' ));

        expect( screen.getByText('11') ).toBeTruthy();
    });

    test('Debe decrementar con el botón -1', () => {
        render( <CounterApp3 number={number} /> );
        fireEvent.click(screen.getByText( '-1' ));

        expect( screen.getByText('9') ).toBeTruthy();
    });

    test('Debe funcionar el botón reset', () => {
        // Sujeto de prueba
        render( <CounterApp3 number={ number } /> );

        //Estímulo
        fireEvent.click(screen.getByText( '-1' ));
        fireEvent.click(screen.getByText( '-1' ));
        fireEvent.click(screen.getByText( '-1' ));
        // fireEvent.click(screen.getByText( 'Reset' ));
        fireEvent.click(screen.getByRole('button', { name: 'btn-reset' }));

        // Aserción
        expect( screen.getByText(`${ number }`) ).toBeTruthy();
    });
});