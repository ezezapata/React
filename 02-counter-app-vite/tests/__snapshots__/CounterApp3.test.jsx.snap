// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en CounterApp3 Debe hacer match con el snapshot. 1`] = `
<div>
  <h1>
    CounterApp 3
  </h1>
  <p>
     
    10
     
  </p>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
</div>
`;
